### PR TITLE
feat: [CDS-45642]: ECS NG S3 Store Feature

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
## Describe your changes
This is new feature in ECS
Amazon S3 buckets can be used as Manifest Store  
Test Plan: https://docs.google.com/spreadsheets/d/1WWrY6S9Q4drtVWi-sUu9kaI0hFdfEFukS_nDROzcjpw/edit#gid=0
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)

### CODE REVIEW

**Comment by alluvamsi-harness** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      return responseData;
+    } catch (Exception e) {
+      Exception sanitizedException = ExceptionMessageSanitizer.sanitizeException(e);
+      log.error("Exception in S3 Fetch Files Task", sanitizedException);
+      throw new TaskNGDataException(
+          UnitProgressDataMapper.toUnitProgressData(commandUnitsProgress), sanitizedException);
+    }
+  }
+
+  private DelegateResponseData getEcsS3CommandResponse(
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java:104]: These methods need to be moved to a Handler calss.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java [hunk 2]
```diff
@@ -77,6 +80,35 @@ public GitStoreDelegateConfig getGitStoreDelegateConfigForRunTask(
...
+    ecsManifestsOutcome.forEach(manifest -> {
+      if (manifest.getStore().getKind() == ManifestStoreType.S3) {
+        isS3Manifest.set(true);
+      }
+    });
+    return isS3Manifest.get();
+  }
+
+  public boolean checkForGitManifest(List<ManifestOutcome> ecsManifestsOutcomes) {
+    AtomicBoolean isGitManifest = new AtomicBoolean(false);
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:94]: Any specific reason why we are using atomic boolean here?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java [hunk 3]
```diff
@@ -77,6 +80,35 @@ public GitStoreDelegateConfig getGitStoreDelegateConfigForRunTask(
         gitStoreConfig, connectorDTO, manifestOutcome, gitStoreConfig.getPaths().getValue(), ambiance);
   }
 
+  public boolean checkForS3Manifest(List<ManifestOutcome> ecsManifestsOutcome) {
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:83]: We can use better method naming here. 'checkForS3Manifest' can be called 'isAnyS3Manifest'

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import io.harness.pms.sdk.core.steps.io.PassThroughData;
+
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.data.annotation.TypeAlias;
+
+@OwnedBy(HarnessTeam.CDP)
+@Value
+@Builder
+@TypeAlias("ecsGitFetchPassThroughData")
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java:23]: It should be 'ecsS3FetchPassThroughData'?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        ecsS3ScalingPolicyContents.add(fetchS3ManifestsContent(ecsS3FetchFileConfig, executionLogCallback));
+      }
+    }
+
+    executionLogCallback.saveExecutionLog(
+        color(format("%nFetched all S3 manifests successfully..%n"), LogColor.White, LogWeight.Bold), INFO,
+        CommandExecutionStatus.SUCCESS);
+
+    return EcsS3FetchResponse.builder()
+        .taskStatus(TaskStatus.SUCCESS)
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java:146]: When are we setting Task Status as 'FAILURE'.? I don't see try catch as well.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java (hunk 2 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:94]: no reason changing it

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java [hunk 6]
```diff
@@ -77,6 +80,35 @@ public GitStoreDelegateConfig getGitStoreDelegateConfigForRunTask(
...
+    AtomicBoolean isS3Manifest = new AtomicBoolean(false);
```
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:84]: no reason changing it

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java (hunk 1 above)
Comment (alluvamsi-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java:104]: done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:84]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java (hunk 2 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:94]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java [hunk 7]
```diff
@@ -77,6 +80,35 @@ public GitStoreDelegateConfig getGitStoreDelegateConfigForRunTask(
...
+    ecsManifestsOutcome.forEach(manifest -> {
+      if (manifest.getStore().getKind() == ManifestStoreType.S3) {
+        isS3Manifest.set(true);
+      }
+    });
+    return isS3Manifest.get();
+  }
+
+  public boolean checkForGitManifest(List<ManifestOutcome> ecsManifestsOutcomes) {
```
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:93]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java (hunk 3 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepUtils.java:83]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java (hunk 4 above)
Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java:23]: Done

**Comment by USB-05** [comment]
trigger sswc

**Comment by USB-05** [comment]
trigger sswc

**Comment by alluvamsi-harness** [comment]
trigger ti3

**Comment by alluvamsi-harness** [comment]
trigger feature-build

**Review by PragyeshMishra01 — CHANGES_REQUESTED** [review]
> Reviewed 8 of 28 files at r1, 21 of 22 files at r2, 1 of 1 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java` line 429 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH95itH0RZ1YWXph6KH:-NH95itH0RZ1YWXph6KI:b8lmbzt) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L429)):*
> ```Java
>         .encryptedDataDetails(
>             k8sEntityHelper.getEncryptionDataDetails(awsConnectorDTO, AmbianceUtils.getNgAccess(ambiance)))
>         .build();
> ```

why are we using k8sEntityHelper class here?
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java` line 1142 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9HQXh3Xyb4UNJHC7f:-NH9HQXh3Xyb4UNJHC7g:b-8t39h5) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java#L1142)):*
> ```Java
>     if (CollectionUtils.isNotEmpty(ecsOtherStoreContent.getEcsScalableTargetManifestContentList())) {
>       ecsScalableTargetFileContentList.addAll(ecsOtherStoreContent.getEcsScalableTargetManifestContentList());
>     }
> ```

In case of ecsOtherStoreContent, don't we need to render expressions?
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java` line 27 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9K7ov8BAMtoPrpWJ9:-NH9K7ov8BAMtoPrpWJA:b-s0spq5) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java#L27)):*
> ```Java
> public class EcsS3FetchPassThroughData implements PassThroughData {
>   String otherStoreTargetGroupArnKey;
>   InfrastructureOutcome infrastructureOutcome;
> ```

what does otherStoreTargetGroupArnKey mean in terms of naming?  How is it depending on store? 
I am seeing that we have used otherStore prefix for some variables.

**Review by alluvamsi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1 change requests. 8. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java` line 429 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH95itH0RZ1YWXph6KH:-NH9WoJW2VDmse_deQPt:b-a9q4c4) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L429)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

why are we using k8sEntityHelper class here?
</blockquote></details>

working
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java` line 1142 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9HQXh3Xyb4UNJHC7f:-NH9WVNo94oyMZPHCITx:by3058p) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java#L1142)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

In case of ecsOtherStoreContent, don't we need to render expressions?
</blockquote></details>

No need they are the otherStoreContents(Git,Harness) which are already rendered 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java` line 27 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9K7ov8BAMtoPrpWJ9:-NH9WyE0Ai4f5UrzMES1:b9g4gsl) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java#L27)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

what does otherStoreTargetGroupArnKey mean in terms of naming?  How is it depending on store? 
I am seeing that we have used otherStore prefix for some variables. 
</blockquote></details>

The otherStore in S3 means the data which they are carrying is from Git or Harness Store
eg: In EcsS3FetchPassThroughData the otherStoreTargetArnKey is either from the Service Definition of Git or Harness Store but not from S3

**Review by PragyeshMishra01 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java` line 1142 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9HQXh3Xyb4UNJHC7f:-NH9_GvG3hKGUFT5A8Qa:b-y8hgn1) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/EcsStepCommonHelper.java#L1142)):*
<details><summary><i>Previously, alluvamsi-harness (Allu Vamsi Vishal) wrote…</i></summary><blockquote>

No need they are the otherStoreContents(Git,Harness) which are already rendered 
</blockquote></details>

Okay, then its good.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java` line 27 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9K7ov8BAMtoPrpWJ9:-NH9Zsd634ZYuYS_ij7d:b151596) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java#L27)):*
<details><summary><i>Previously, alluvamsi-harness (Allu Vamsi Vishal) wrote…</i></summary><blockquote>

The otherStore in S3 means the data which they are carrying is from Git or Harness Store
eg: In EcsS3FetchPassThroughData the otherStoreTargetArnKey is either from the Service Definition of Git or Harness Store but not from S3
</blockquote></details>

if they belong to harness or git store, then why are we putting them inside EcsS3FetchPassThroughData  class?

**Review by alluvamsi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1 change requests. 8. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java` line 27 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH9K7ov8BAMtoPrpWJ9:-NH9rcPv0RjbTwR9WTrG:b-v0jd30) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/ecs/beans/EcsS3FetchPassThroughData.java#L27)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

if they belong to harness or git store, then why are we putting them inside EcsS3FetchPassThroughData  class?
</blockquote></details>

The only way to pass the harness , gits files to EcsCommand Task is through the S3 Task so they are passed as passThrough
PassThroughData contains the data which we will use in future task but may not be used in current task if a right
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java` line 146 at r1](https://reviewable.io/reviews/harness/harness-core/40149#-NGtnDSr6hvxP3_d5CQG-r1-146:-NH9rRKk7zFlyK7qI9I6:b-896fjx) ([raw file](https://github.com/harness/harness-core/blob/b69cd7845c64897a6fbfaeec121381b425d112c0/930-delegate-tasks/src/main/java/io/harness/delegate/task/ecs/EcsS3FetchTask.java#L146)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

When are we setting Task Status as 'FAILURE'.? I don't see try catch as well.
</blockquote></details>

Done

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 8. unresolved discussions.

**Review by PragyeshMishra01 — APPROVED** [review]

**Comment by alluvamsi-harness** [comment]
trigger ft1
trigger ft0

**Comment by alluvamsi-harness** [comment]
trigger ft1

File: 125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java [hunk 8]
```diff
@@ -415,6 +420,53 @@ public GitFetchFilesConfig getGitFetchFilesConfigFromBuilder(String identifier,
...
+        .paths(s3StoreConfig.getPaths().getValue())
+        .awsConnector((AwsConnectorDTO) awsConnectorDTO.getConnectorConfig())
+        .encryptedDataDetails(getEncryptionDataDetails(awsConnectorDTO, AmbianceUtils.getNgAccess(ambiance)))
+        .build();
+  }
+
+  public List<EncryptedDataDetail> getEncryptionDataDetails(
+      @Nonnull ConnectorInfoDTO connectorDTO, @Nonnull NGAccess ngAccess) {
+    switch (connectorDTO.getConnectorType()) {
+      case AWS:
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java:437]: @alluvamsi-harness lets discuss on why this code is needed. is it method not present already?

**Comment by alluvamsi-harness** [comment]
trigger codebasehash

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — COMMENTED** [review]
> Reviewed 1 of 28 files at r1, 1 of 22 files at r2, 1 of 8 files at r6, 1 of 34 files at r8, 30 of 34 files at r11, 6 of 6 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 38 file(s) to review. 8. unresolved discussions. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 38 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

**Comment by alluvamsi-harness** [comment]
trigger ti0

**Comment by alluvamsi-harness** [comment]
trigger ti1

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 22 files at r2, 8 of 34 files at r11, 1 of 6 files at r12, 1 of 7 files at r13, 1 of 1 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 30 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Comment by alluvamsi-harness** [comment]
trigger smartchecks

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 6 of 7 files at r13, 1 of 1 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 28 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by PragyeshMishra01 — COMMENTED** [review]
> Reviewed 1 of 8 files at r6, 22 of 34 files at r11, 6 of 6 files at r12, 6 of 7 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1. unresolved discussions.

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 3 of 34 files at r11, 1 of 1 files at r14.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1. unresolved discussions.

**Review by alluvamsi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 2. unresolved discussions. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java` line 429 at r4](https://reviewable.io/reviews/harness/harness-core/40149#-NH95itH0RZ1YWXph6KH:-NIAfFUj-hPqs_r8xmGz:b9lmg9q) ([raw file](https://github.com/harness/harness-core/blob/680a9b528f1cbee74ef43f8a285a60da855eca74/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L429)):*
<details><summary><i>Previously, alluvamsi-harness (Allu Vamsi Vishal) wrote…</i></summary><blockquote>

working
</blockquote></details>

Not changing it as discussed in call

**Review by alluvamsi-harness — COMMENTED** [review]
> Reviewed 1 of 28 files at r1, 1 of 8 files at r6, 22 of 34 files at r11, 6 of 6 files at r12, 6 of 7 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 2. unresolved discussions.

**Review by PragyeshMishra01 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 8 of 9 files at r15.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 9 file(s) to review. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 9 file(s) to review. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 28 files at r1, 1 of 8 files at r6, 17 of 34 files at r11, 5 of 6 files at r12, 1 of 9 files at r15.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 8 of 9 files at r15, 2 of 2 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 8 of 9 files at r15, 1 of 2 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: 1 file(s) to review.

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 1 of 2 files at r16.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 2 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40149)* status: :shipit: complete! ✓

**Pull request merged**